src/Synapse/Synapse/Commands/ManagementCommands/Auditing/SynapseSqlPoolAuditCmdlet.cs

```diff
@@ -85,6 +85,11 @@ public class SynapseSqlPoolAuditCmdlet : AzureSynapseSqlPoolManagementCmdletBase
 
         protected override SqlPoolAuditModel GetEntity()
         {
+            if (string.IsNullOrEmpty(this.ResourceGroupName))
```

Comment:
Need to add check for if `WorkspaceName` is not null or empty, because the `WorkspaceName` is a mandatory parameter only in `WorkspaceParameterSet` parameter set.

Reply:
move the judgement to the end might the problem, @wonner  is that ok?